app/code/Magento/Quote/Model/QuoteManagement.php

```diff
@@ -298,22 +298,30 @@ public function assignCustomer($cartId, $customerId, $storeId)
             );
         }
         try {
-            $this->quoteRepository->getForCustomer($customerId);
-            throw new StateException(
-                __("The customer can't be assigned to the cart because the customer already has an active cart.")
-            );
-        // phpcs:ignore Magento2.CodeAnalysis.EmptyBlock
+            $customerActiveQuote = $this->quoteRepository->getForCustomer($customerId);
         } catch (\Magento\Framework\Exception\NoSuchEntityException $e) {
+            /** This exception appear when customer have no active cart*/
+            $customerActiveQuote = false;
         }
 
+        if ($customerActiveQuote) {
+            /** Merge carts */
+            $quote->merge($customerActiveQuote);
```

Comment:
Please, move these lines to the try block

Reply:
@sidolov , thank you for comment. Update implemented.